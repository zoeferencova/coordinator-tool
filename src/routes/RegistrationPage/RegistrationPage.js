import React from 'react';
import { Link } from 'react-router-dom';
import RegistrationForm from '../../components/RegistrationForm/RegistrationForm';
import { Button } from '../../components/Utils/Utils'

import styles from '../LoginPage/LoginPage.module.css'

export default class RegistrationPage extends React.Component {
    static defaultProps = {
        history: {
            push: () => {},
        },
    }

    handleRegistrationSuccess = user => {
        const { history } = this.props;
        history.push('/login')
    }
    
    render() {
        return (
            <div className={styles.main}>
                <h1 className={styles.title}>Sign up</h1>
                <div className={styles.container}>
                    <RegistrationForm onRegistrationSuccess={this.handleRegistrationSuccess}  />
                    <p className={styles.sub}>Already have an account? <Link to='/login'>Sign in</Link></p>
                </div>
            </div>
        )
    }
}