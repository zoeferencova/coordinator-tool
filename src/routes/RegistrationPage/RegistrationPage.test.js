import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import RegistrationPage from './RegistrationPage'

describe(`RegistrationPage component`, () => {
  it('renders a .RegistrationPage by default', () => {
    const wrapper = shallow(<RegistrationPage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})