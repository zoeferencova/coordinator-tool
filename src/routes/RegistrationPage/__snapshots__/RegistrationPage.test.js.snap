// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationPage component renders a .RegistrationPage by default 1`] = `
<div
  className="main"
>
  <h1
    className="title"
  >
    Sign up
  </h1>
  <div
    className="container"
  >
    <RegistrationForm
      onRegistrationSuccess={[Function]}
    />
    <p
      className="sub"
    >
      Already have an account? 
      <Link
        to="/login"
      >
        Sign in
      </Link>
    </p>
  </div>
</div>
`;
